### ISSUE TITLE
Backend sorting on queries

### ISSUE THREAD

**Issue by ToxicMushroom** [kanidm/kanidm#3071]
### Is your feature request related to a problem? Please describe
For the UI it would be nice if lists of entries could be sorted on some attribute so the order stays predictable for users.
E.g. coming back to the same web-page with a completely different order of users is not nice imo.

### Describe the solution you'd like
As noted by @Firstyear in #3019 we could add sorting functionality to the backend

### Describe alternatives you've considered
Sorting the entries after they were returned from the db but this wastes performance by undoing the work that was put into creating the unsorted list, and the db might be able to sort more efficiently because it has indexes.

This alternative is even more infeasible for paginated responses.

**Comment by yaleman** [comment]
Sorting options should be a default feature on any interface (REST API, code, cli or web UI) that responds with more than one element. That doesn't mean sort by default but as an option to do so in a stable way.

**Comment by Firstyear** [comment]
Yep, I just wanted the ticket open to track me needing to add it :)

**Comment by yaleman** [comment]
Yeah, just dumping my own thoughts into it to make sure it's called out in detail too 😄
